### ISSUE TITLE
Configurable cache namespace

### ISSUE THREAD

**Issue by Ksnz** [micronaut-projects/micronaut-redis#231]
### Feature description

If we have different applications with same cache name and same Redis instance it may cause incorrect behavior of mixing different caches.
I think it would be nice to have an option for cache config like a `namespace`.
Or using application name as namespace prefix by default.

**Comment by jfsanchez91** [comment]
I guess you can set manually this prefix, just use the application name as a prefix for your cache names.

**Comment by Ksnz** [comment]
Yes but what if cache configuration is hardcoded as part of some business logic?
Like
```
@Named(SomeStuffCacheConfiguration.CACHE_NAME)
@ConfigurationProperties(CacheConfiguration.PREFIX + "." + SomeStuffCacheConfiguration.CACHE_NAME)
public class SomeStuffCacheConfiguration extends CacheConfiguration [
public static final String CACHE_NAME = "some-stuff-cache";
```

**Comment by ojebari-ma** [comment]
Can you please confirm that this is still an issue in the latest version of Micronaut?
